tests/ijby/test-cut.py

```diff
@@ -0,0 +1,213 @@
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+from datatable import dt, stype, f, cut
+from tests import assert_equals
+import pytest
+import random
+import math
+
+
+#-------------------------------------------------------------------------------
+# cut()
+#-------------------------------------------------------------------------------
+
+def test_cut_error_noargs():
+    msg = r"Function cut\(\) requires one positional argument, but none were given"
+    with pytest.raises(TypeError, match=msg):
+        cut()
+
+
+def test_cut_error_wrong_column_types():
+    DT = dt.Frame([[1, 0], ["1", "0"]])
+    msg = r"cut\(\) can only be applied to numeric columns, instead column 1 has an stype: str32"
+    with pytest.raises(TypeError, match=msg):
+        cut(DT)
+
+
+def test_cut_error_negative_bins():
+    msg = "Integer value cannot be negative"
+    DT = dt.Frame(range(10))
+    with pytest.raises(ValueError, match=msg):
+        cut(DT, -10)
+
+
+def test_cut_error_negative_bins_list():
+    msg = "Integer value cannot be negative"
+    DT = dt.Frame([[3, 1, 4], [1, 5, 9]])
+    with pytest.raises(ValueError, match=msg):
+        cut(DT, [10, -1])
+
+
+def test_cut_error_inconsistent_bins():
+    msg = ("When nbins is a list or a tuple, its length must be the same as "
+    	   "the number of columns in the frame/expression, i.e. 2, instead got: 1")
+    DT = dt.Frame([[3, 1, 4], [1, 5, 9]])
+    with pytest.raises(ValueError, match=msg):
+        cut(DT, [10])
+
+
+def test_cut_error_wrong_right():
+    msg = "Expected a boolean, instead got <class 'int'>"
+    DT = dt.Frame(range(10))
+    with pytest.raises(TypeError, match=msg):
+        cut(DT, right_closed = 1492)
+
+
+def test_cut_error_groupby():
+    msg = r"cut\(\) cannot be used in a groupby context"
+    DT = dt.Frame(range(10))
+    with pytest.raises(NotImplementedError, match=msg):
+        DT[:, cut(f[0]), f[0]]
+
+
+def test_cut_empty_frame():
+	DT_cut = cut(dt.Frame())
+	assert_equals(DT_cut, dt.Frame())
```

Comment:
Looks like the files uses tabs mixed with spaces. There is "convert indentation to spaces" function in Sublime.

Reply:
Yeah, I had tabsize equal two in this file for some reason :/